.github/workflows/scenarios-permian.yml

```diff
@@ -0,0 +1,120 @@
+# Run all kickstart tests for all active branches in Permian
+name: Daily run in Permian
+on:
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  scenario:
+    name: Scenario in Permian
+    runs-on: [self-hosted, kstest-permian]
+    strategy:
+      matrix:
+        scenario: [daily-iso, rhel8, rhel9]
+      fail-fast: false
+
+    # these settings depend on the infrastructure; on upshift ocp-master-xxl they take about 4 hours
+    timeout-minutes: 480
+    env:
+      TEST_JOBS: 16
+      GITHUB_TOKEN: /home/github/github-token
+
+    steps:
+      # self-hosted runners don't do this automatically; also useful to keep stuff around for debugging
+      # need to run sudo as the launch script and the container create root/other user owned files
+      - name: Clean up previous run
+        run: |
+          sudo podman ps -q --all --filter='ancestor=kstest-runner' | xargs -tr sudo podman rm -f
+          sudo podman volume rm --all || true
+          sudo rm -rf *
+
+      - name: Clone repository
+        uses: actions/checkout@v2
+        with:
+          path: kickstart-tests
+
+      - name: Generate test cases
+        working-directory: ./kickstart-tests
+        run: scripts/generate-testcases.py -t ./testlib/test_cases/kstest-template.tc.yaml.j2 . -o ./testlib/test_cases
+
+      - name: Clone Permian repositories
+        run: |
+          git clone https://github.com/rhinstaller/permian.git
+          git clone https://github.com/rhinstaller/tclib.git
```

Comment:
I would rather like to use checkout actions instead of clone.

Reply:
I cautiously agree, but only because of consistency.

I see no reason to consider the actions "better" or "correct". The whole workflow is de facto a shell script with some additional stuff instead of comments, so perhaps it's actually the actions that are out of place. After all, the only other actions we consistently use in rhinstaller org are related to the platform (api calls, upload files).

Reply:
Updated with the action.